xbmc/addons/AddonManager.cpp

```diff
@@ -659,16 +670,16 @@ static void ResolveDependencies(const std::string& addonId, std::vector<std::str
   }
 }
 
-bool CAddonMgr::DisableAddon(const std::string& id)
+bool CAddonMgr::DisableAddon(const std::string& id, const AddonDisabledReason& disabledReason)
 {
   CSingleLock lock(m_critSection);
   if (!CanAddonBeDisabled(id))
     return false;
   if (m_disabled.find(id) != m_disabled.end())
     return true; //already disabled
-  if (!m_database.DisableAddon(id))
+  if (!m_database.DisableAddon(id, disabledReason))
     return false;
-  if (!m_disabled.insert(id).second)
+  if (!m_disabled.insert({id, disabledReason}).second)
```

Comment:
Maybe there to use ".emplace(...)" instead of insert

Reply:
Why use emplace?

Reply:
@ksooo mean always to use it, by my changes. Think in work a bit better.

@ksooo what are in work there better for emplace, instead of insert?

Reply:
On a map I think the it’s only that one replaces the element and the other doesn’t.

Reply:
I just read up on it. It’s much better to use. Thanks!